src/components/CancelButton/CancelButton.js

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* eslint-disable react/no-array-index-key */
+import React, { Component } from 'react';
+
+import { Icon, Modal } from 'carbon-components-react';
+
+import './CancelButton.scss';
+
+class CancelButton extends Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.handleShow = this.handleShow.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+
+    this.state = {
+      show: false
```

Comment:
maybe `showDialog`?

Also wondering if we should only display the cancel button on hover / focus... we can revisit as part of the design updates / when adding other actions.